rust/ql/src/queries/security/CWE-825/AccessAfterLifetime.ql

```diff
@@ -28,6 +28,15 @@ module AccessAfterLifetimeConfig implements DataFlow::ConfigSig {
   predicate isSink(DataFlow::Node node) { node instanceof AccessAfterLifetime::Sink }
 
   predicate isBarrier(DataFlow::Node barrier) { barrier instanceof AccessAfterLifetime::Barrier }
+
+  predicate observeDiffInformedIncrementalMode() { any() }
+
+  Location getASelectedSourceLocation(DataFlow::Node source) {
+    exists(Variable target, DataFlow::Node sink | result = target.getLocation() |
+      isSink(sink) and
+      AccessAfterLifetime::dereferenceAfterLifetime(source, sink, target)
```

Comment:
Should this not check the `flowPath` exists as well?

I guess it may not be required for correctness actually ... but `dereferenceAfterLifetime` is `bindingset[source, sink]` for performance reasons, I'm a little bit concerned that not checking the `flowPath` here may greatly increase the amount we compute for it.

Reply:
Unfortunately, we can't check `flowPath` because that would lead to non-monotonic recursion (`getASelected…Location` is negated somewhere in `flowPath`).

DCA shows a slight (probably within noise range) increase in `neon-empty-diff`, so it's possible this has a negative impact.

Maybe an alternative is to include the rest of the clauses from the `where`, which might reduce the search space before calling `dereferenceAfterLifetime`.

Reply:
Yeah, the `unsafe` block check in particular may narrow things down a lot.